pkg/jobs/registry.go

```diff
@@ -340,6 +363,42 @@ func (r *Registry) maybeCancelJobs(ctx context.Context, nl NodeLiveness) {
 	}
 }
 
+func (r *Registry) cleanupOldJobs(ctx context.Context) error {
+	old := timeutil.Now().Add(-1 * gcSetting.Get(&r.settings.SV))
+
+	const stmt = `SELECT id, payload FROM system.jobs WHERE status IN ($1, $2, $3) AND created < $4 ORDER BY created LIMIT 100`
```

Comment:
Overall LGTM. I have light reservations about the LIMIT 100 and gc interval. If more than 100 jobs enter a finalized state per hour we won't catch up. Is there a compelling reason we need the LIMIT? Can it be increased to 1000 or just removed?

Reply:
made it 1k just in case